### PR TITLE
[SW-1211] Automatically download H2O Wheel during build

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

**Comment by jakubhava** [comment]
Waiting for tests

File: doc/src/site/sphinx/devel/build.rst [hunk 1]
```diff
@@ -3,23 +3,19 @@ Build Sparkling Water
...
-
 - To avoid running tests, use the ``-x test -x integTest`` or the ``-x check`` option.
 
 - To build only a specific module, use, for example, ``./gradlew :sparkling-water-examples:build``.
 
 - To build and test a specific module, use, for example, ``./gradlew :sparkling-water-examples:check``.
 
 Sparkling Water SUBST_SW_VERSION is built with Scala 2.11.
+
+Note: If you would like to build against custom H2O Python package, specify H2O_HOME environment variable. The variable
```
Comment (mn-mikke) [inline doc/src/site/sphinx/devel/build.rst]: \``H2O_HOME``?

File: doc/src/site/sphinx/devel/build.rst [hunk 2]
```diff
@@ -3,23 +3,19 @@ Build Sparkling Water
...
+should point to root directory of H2O-3 repository. This is mainly used for integration testing with H2O-3.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/devel/build.rst]: the root directory?

File: jenkins/sparklingWaterPipeline.groovy [hunk 3]
```diff
@@ -178,10 +176,6 @@ def prepareSparklingWaterEnvironment() {
                         ${getGradleCommand(config)} -q :sparkling-water-examples:build -x check -PdoExtend extendJar
                     fi
                 else
```
Comment (mn-mikke) [inline jenkins/sparklingWaterPipeline.groovy]: ```else if``` at one line?

File: py/build.gradle [hunk 4]
```diff
@@ -17,6 +17,7 @@ ext {
     pkgDir = file("$buildDir/pkg")
     distDir = file("$buildDir/dist")
     condaDir = file("$buildDir/conda/h2o_pysparkling_${sparkMajorVersion}")
+    wheelFile = file("${rootDir.toString()}/.gradle/wheels/h2o-${h2oMajorVersion}.${h2oBuild}-py2.py3-none-any.whl")
```
Comment (mn-mikke) [inline py/build.gradle]: Is ```.toString()``` necessary here?

File: py/build.gradle [hunk 5]
```diff
@@ -204,15 +199,13 @@ def copyH2OFromH2OWheel(String h2oPythonWheel) {
...
         // if both properties are set, give precedence to H2O_HOME
-        if (H2O_HOME != null && H2O_PYTHON_WHEEL != null) {
-            copyH2OFromH2OHome(H2O_HOME)
-        } else if (H2O_HOME != null) {
+        if (H2O_HOME != null) {
             copyH2OFromH2OHome(H2O_HOME)
-        } else if (H2O_PYTHON_WHEEL != null) {
-            copyH2OFromH2OWheel(H2O_PYTHON_WHEEL)
+        } else {
+            copyH2OFromH2OWheel(H2O_PYTHON_WHEEL.toString())
```
Comment (mn-mikke) [inline py/build.gradle]: Is ```.toString()``` necessary here?

File: doc/src/site/sphinx/devel/build.rst (hunk 2 above)
Comment (jakubhava) [reply doc/src/site/sphinx/devel/build.rst]: 👍

File: py/build.gradle (hunk 5 above)
Comment (jakubhava) [reply py/build.gradle]: Yup, `H2O_PYTHON_WHEEL` is `File`. I made it more obvious. thx!

File: py/build.gradle (hunk 4 above)
Comment (jakubhava) [reply py/build.gradle]: nope, good point

Comment (jakubhava) [reply py/build.gradle]: it will resolve to string automatically

File: py/build.gradle (hunk 5 above)
Comment (jakubhava) [reply py/build.gradle]: Actually no even here, we can pass in file directly

**Comment by jakubhava** [comment]
Thanks for review @mn-mikke. Incorporated the changes in, let's wait for tests

**Pull request merged**